libbeat/outputs/outputs.go

```diff
@@ -34,7 +36,7 @@ type Client interface {
 	// Using Retry/Cancelled a client can return a batch of unprocessed events to
 	// the publisher pipeline. The publisher pipeline (if configured by the output
 	// factory) will take care of retrying/dropping events.
-	Publish(publisher.Batch) error
+	Publish(context.Context, publisher.Batch) error
```

Comment:
This is a big change. When passing a context, is it only for tracing, or do we expect Deadline, Done, and Err to be used as well? In the later case error handling might need to be improved further to handle/ignore context.Cancelled. E.g. we do not want use Batch.Retry (decrease counter and eventually drop) on cancel, but Batch.Cancelled (keep TTL counter and rescheduled batch with another output).

Reply:
I had not intended to introduce cancellation/deadlines/etc as part of this change, but this could enable that as part of a separate effort.

Reply:
Not sure how feasible this would be :)

In this case can you please update the code comment to clarify the usage of the context passed?